classes/Reputation.php

```diff
@@ -0,0 +1,323 @@
+            'outlaw_title' => 'Vagabond',
+            'min_rep' => 0,
+            'weekly_cap' => 500,
+            'base_pvp_rep_reward' => 1,
+            'base_decay' => 0,
+        ],
+        2 => [
+            'title' => 'Villager',
+            'outlaw_title' => 'Thief',
+            'min_rep' => 500,
+            'weekly_cap' => 500,
+            'base_pvp_rep_reward' => 1,
+            'base_decay' => 300,
+        ],
+        3 => [
+            'title' => 'Well-Known Villager',
+            'outlaw_title' => 'Infamous',
+            'min_rep' => 1000,
+            'weekly_cap' => 500,
+            'base_pvp_rep_reward' => 1,
+            'base_decay' => 400,
+        ],
+        4 => [
+            'title' => 'Respected Villager',
+            'outlaw_title' => 'Outlaw',
+            'min_rep' => 1750,
+            'weekly_cap' => 600,
+            'base_pvp_rep_reward' => 2,
+            'base_decay' => 450,
+        ],
+        5 => [
+            'title' => 'Shinobi',
+            'outlaw_title' => 'Rogue',
+            'min_rep' => 2500,
+            'weekly_cap' => 600,
+            'base_pvp_rep_reward' => 2,
+            'base_decay' => 500,
+        ],
+        6 => [
+            'title' => 'Respected Shinobi',
+            'outlaw_title' => 'Infamous Rogue',
+            'min_rep' => 3500,
+            'weekly_cap' => 600,
+            'base_pvp_rep_reward' => 2,
+            'base_decay' => 550,
+        ],
+        7 => [
+            'title' => 'Elite Shinobi',
+            'outlaw_title' => 'Assassin',
+            'min_rep' => 5000,
+            'weekly_cap' => 700,
+            'base_pvp_rep_reward' => 3,
+            'base_decay' => 600,
+        ],
+        8 => [
+            'title' => 'Master Shinobi',
+            'outlaw_title' => 'Master Assassin',
+            'min_rep' => 7500,
+            'weekly_cap' => 800,
+            'base_pvp_rep_reward' => 3,
+            'base_decay' => 650,
+        ],
+        9 => [
+            'title' => 'Legendary Shinobi',
+            'outlaw_title' => 'Legendary Assassin',
+            'min_rep' => 10000,
+            'weekly_cap' => 900,
+            'base_pvp_rep_reward' => 3,
+            'base_decay' => 750,
+        ]
+    ];
+
+    const ARENA_MISSION_CD = 60;
+    const MISSION_GAINS = [
+        Mission::RANK_D => 1,
+        Mission::RANK_C => 1,
+        Mission::RANK_B => 2,
+        Mission::RANK_A => 3,
+        Mission::RANK_S => 4
+    ];
+    const SPECIAL_MISSION_REP_GAINS = [
+        SpecialMission::DIFFICULTY_EASY => 1,
+        SpecialMission::DIFFICULTY_NORMAL => 1,
+        SpecialMission::DIFFICULTY_HARD => 2,
+        SpecialMission::DIFFICULTY_NIGHTMARE => 3,
+    ];
+    const DAILY_TASK_GAINS = [
+        DailyTask::DIFFICULTY_EASY . '_' . DailyTask::ACTIVITY_ARENA => 1,
+        DailyTask::DIFFICULTY_MEDIUM . '_' . DailyTask::ACTIVITY_ARENA => 3,
+        DailyTask::DIFFICULTY_HARD . '_' . DailyTask::ACTIVITY_ARENA => 5,
```

Comment:
Instead of string interpolation, what if we made this a nested array? Seems a little cleaner and easier to do static type checking
```suggestion
    const DAILY_TASK_GAINS = [
        DailyTask::DIFFICULTY_EASY => [
            DailyTask::ACTIVITY_ARENA => 1,
        ],
        DailyTask::DIFFICULTY_MEDIUM => [
            DailyTask::ACTIVITY_ARENA => 3,
        ],
        DailyTask::DIFFICULTY_HARD => [
            DailyTask::ACTIVITY_ARENA => 5,
        ],
```